src/pile.sol

```diff
@@ -226,34 +123,29 @@ contract Pile is DSNote, TitleOwned {
         return loans_[loan].balance;
     }
 
+    function collect(uint loan) public {
+        beans.drip(loans_[loan].fee);
+    }
+
     // repay() a certain amount of token from the user to the Pile
     function repay(uint loan, uint wad) public owner(loan) note {
         // moves currency from usr to pile and reduces debt
         require(loans_[loan].balance == 0,"before repay loan needs to be withdrawn");
-        collect(loan);
+        uint fee = loans_[loan].fee;
+        beans.drip(fee);
+        uint debt = beans.debtOf(loan, fee);
 
         // only repay max loan debt
-        uint debt = debtOf(loan);
         if (wad > debt) {
             wad = debt;
         }
 
         tkn.transferFrom(msg.sender, address(this), wad);
-
-        uint chi = getChi(loan);
-        uint pie_ = rdiv(wad, chi);
-        loans_[loan].pie = sub(loans_[loan].pie, pie_);
-
-        decDebt(loans_[loan].fee, wad);
+        beans.decLoanDebt(loan, fee, wad);
         tkn.approve(lender, wad);
```

Comment:
I think we can remove that line for Naxos. The pile should allow the `trancheManager` full control over the currency in pile.
I am not sure about the best pattern yet

Reply:
let's discuss this face to face